### ISSUE TITLE
Is this project really open source?

### ISSUE THREAD

**Issue by Ashymad** [AoTTG-2/AoTTG-2#211]
In the README it is stated that "From here, we are developing an open-source project known as AoTTG 2" and the license file in this repo contains the GPLv3.
But as can be seen in the repo's history the code is based on RiceCake's RC Mod source ee23cca06f9d3273717fd099f32d3f8052d9b08a, which is in turn based on the original game by Feng Lee. Do you have permission from RiceCake and Feng to relicense their work under the GPLv3?
If the answer is "no" then this project is not open source, most of it is proprietary and used without permission. And unless all of Feng's and RiceCake's work is recreated from scratch it will be. As such contributing to this repo is legally dubious.
If the answer is "yes" then please place this info along with relevant screenshots (with Feng's and RiceCake's approval) in the README, this is very important information to future contributors.

**Comment by Lithrun** [comment]
Yes, the project is open source. All our work is publicly available, and will remain publicly available after the games release as well. When AoTTG2 development started, it used the RC (RiceCake's) mod as a base, as a starting point if you may want to call it like that.

The developer of the RC mod is within our development team itself. Unfortunately, we do not have explicit permission from FengLee however. This is because it seems impossible to get into contact with him. FengLee has been gone since 2016 sadly.

Internally there are still quite some leftovers of Fengs / RCs code, but eventually this will all get refactored and improved to AoTTG2 quality. The state of the code of AoTTG is quite dramatic and has a very low maintainability standard.

TLDR: RiceCake is on the dev team, most code will get refactored. The more time progresses, the less AoTTG code there is left, and the more AoTTG2 code there is.

**Comment by Ashymad** [comment]
Thanks for the answer!

Although you say that the code will be refactored, legally it would be considered derived work and as such can't be relicensed. I know that it's impossible to contact Feng Lee currently and I hope that he will eventually approve of this work, but I would warn contributors that currently their work is legally worthless as they have no right to modify Feng's code (I am pretty sure that even the act of decompiling Feng's aot binary would be considered illegal).

I will leave it in your hands as to how do you want to handle this. I hope the project will succeed, but calling it Open Source is not right. Yeah the code is publicly available, because you uploaded it to GitHub, but you have no right to upload and modify it and it can be taken down at any moment by Feng if he decides so. (Or basically anyone, as you are violating GitHub ToS).

TL;DR: Rewriting is not enough to relicense, you would have to have Feng's permission or recreate everything from scratch (without reading the original code). This repo is not an open source aottg clone, but just a continuation of RC Mod, as illegal as the original mod, just developed in the open.

**Comment by Lithrun** [comment]
The license under this repository is one that we determined is the best one for the project as a whole, to stimulate cooperation and to be publicly available, free of charge to anyone of the community that wants to contribute. The license however, holds no status from a legal perspective, with what you mentioned above.

The legal state of fan projects is debatable in the first place, given the fact that we do not have explicit permission of the copyright holders. AoTTG is illegal on itself. There is always a risk that copyright owners could force us to shutdown. Perhaps Feng could do the same as well. Being able to contact him and get his permission would be great, as then at least the license to the code can be made valid, but again, nobody has been able to do so since 2016, and letting the community die is not an option either. So until we somehow figure out how to contact him, that remains a grey area. We are still trying to contact him however, so hopefully we can follow up on that.

It would definitely suck for the whole community if the AoT copyright owners decide to shut it down, but it is what it is. I'd be surprised if they have never heard of AoTTG, and no action was taken against them, neither was any action taken against other AoT fan games. This could change in the future sadly, but since we are not monetizing anything, and since it is a project by fans for fans, let's hope for the best 👍 

TLDR: We are still trying to contact FengLee to have his explicit permission. Once we have that, all the code itself can safely be licensed under our current license, and then we only have to worry about AoT copyright owners.

**Comment by Lithrun** [comment]
@Ashymad 
The impossible happened, but FengLee replied! We are allowed to work on AoTTG2 as long as it is not used for commercial purposes. We don't have any intend to sell or earn any profits from the game, so that should all be good! We're currently sorting out where to place this information and shortly our main, contributors and FAQ pages should be updated. Thank you for the reminder 👍

Source: https://www.weibo.com/1664859317/J9bZrB0kh

*Closing issue now since permission from both FengLee & RC have been sorted*

**Comment by Ashymad** [comment]
Great! This does however mean that his code is GPL incompatible. You cannot use any Open Source license (as defined by OSI), because all of them require that the code can be used for any purpose (including commercial), instead you should consider using CC BY-NC-SA https://creativecommons.org/licenses/by-nc-sa/4.0/

**Comment by xXXConkerXXx** [comment]
Did Feng even patent his work?
